crates/red_knot_python_semantic/resources/mdtest/generics.md

```diff
@@ -66,3 +57,50 @@ class S[T](Seq[S]): ...  # error: [non-subscriptable]
 
 reveal_type(S)  # revealed: Literal[S]
 ```
+
+## Type params
+
+A PEP695 type variable defines a value of type `typing.TypeVar` with attributes `__name__`,
+`__bounds__`, `__constraints__`, and `__default__` (the latter three all lazily evaluated):
+
+```py
+def f[T, U: A, V: (A, B), W = A, X: A = A1]():
+    reveal_type(T)  # revealed: TypeVar
+    reveal_type(T.__name__)  # revealed: Literal["T"]
+    reveal_type(T.__bound__)  # revealed: None
+    reveal_type(T.__constraints__)  # revealed: tuple[()]
+    reveal_type(T.__default__)  # revealed: NoDefault
+
+    reveal_type(U)  # revealed: TypeVar
+    reveal_type(U.__name__)  # revealed: Literal["U"]
+    reveal_type(U.__bound__)  # revealed: type[A]
```

Comment:
These types could more precisely be class literals; `Literal[A]` instead of `type[A]`. But we want to store them as the instance type they represent, not as their runtime value, because the priority is efficiency of using typevars as type annotations, not introspecting their dunder attributes. So that means this introspection is effectively round-tripping the type through a `.to_instance()` and then back through `.to_meta_type()`. Since a bound can be any kind of type, we have to do this round-trip generically, which means we get back `type[A]` from that round trip, not `Literal[A]`.

With some special-casing of `Type::Instance` we could get `Literal[A]` here, but I don't think it's worth it. Precision of these dunder types matters very little, and `type[]` will be more familiar to users anyway.